### PR TITLE
Preload install animation (SVG)

### LINKED ISSUE #6169: Preload install animation (svg) to avoid a flicker on the very first time
### Describe the problem and steps to reproduce it:

1. Go to a fresh add-on detail page
2. Install an add-on
3. Observe the animation

### What happened?

Sometimes (depending on the network), the SVG does not load the very first time or it is late because the SVG is downloaded when the animation is shown.

### What did you expect to happen?

No rendering issue.

### Anything else we should know?

Preloading the SVG will fix the issue.

### PR DESCRIPTION
**Pull request by willdurand**
Fix #6169

---

This PR merges the `AnimatedIcon` component into the `AMInstallButton`
so that we can preload the SVG properly.

### CODE REVIEW

File: tests/unit/core/components/TestAMInstallButton.js [hunk 1]
```diff
@@ -525,31 +524,31 @@ describe(__filename, () => {
     expect(root.find('.AMInstallButton-button')).toHaveLength(0);
 
     expect(root.find('.AMInstallButton-loading')).toHaveLength(1);
-    expect(root.find(AnimatedIcon)).toHaveLength(1);
+    expect(root.find(Icon)).toHaveLength(1);
```
Comment (rebmullin) [inline tests/unit/core/components/TestAMInstallButton.js]: should we add this here:

`expect(root.find(Icon)).toHaveProp('name', 'loading');`

to show the Icon is in fact loading?

File: tests/unit/core/components/TestAMInstallButton.js [hunk 2]
```diff
@@ -499,20 +498,20 @@ describe(__filename, () => {
   });
 
   it.each([DOWNLOADING, ENABLING, INSTALLING, UNINSTALLING])(
-    'renders an AnimatedIcon when add-on is %s',
+    'renders an Icon when add-on is %s',
```
Comment (rebmullin) [inline tests/unit/core/components/TestAMInstallButton.js]: maybe 'renders a loading Icon when..'?

**Comment by willdurand** [comment]
> I also like the idea of using css animation here if we can:
> 
> #6169 (comment)
> 
> maybe in a follow up issue..?

Yup, that's an interesting idea.

File: tests/unit/core/components/TestAMInstallButton.js (hunk 2 above)
Comment (willdurand) [reply tests/unit/core/components/TestAMInstallButton.js]: ah right

**Pull request merged**